contracts/common/errors/payment_splitter.rs

```diff
@@ -0,0 +1,57 @@
+use super::{
+    AccessControlError,
+    OwnableError,
+    PausableError,
+    ReentrancyGuardError,
+};
+use ink_prelude::string::String;
+
+/// The PaymentSplitter error type. Contract will throw one of this errors.
+#[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum PaymentSplitterError {
+    Custom(String),
+    NoPayees,
+    AccountHasNoShares,
+    AccountIsNotDuePayment,
+    AccountZeroAddress,
+    SharesAreZero,
+    AlreadyHasShares,
+    TransferFailed,
+}
+
+impl From<AccessControlError> for PaymentSplitterError {
+    fn from(access: AccessControlError) -> Self {
+        match access {
+            AccessControlError::MissingRole => PaymentSplitterError::Custom(String::from("AC::MissingRole")),
+            AccessControlError::RoleRedundant => PaymentSplitterError::Custom(String::from("AC::RoleRedundant")),
+            AccessControlError::InvalidCaller => PaymentSplitterError::Custom(String::from("AC::InvalidCaller")),
+        }
+    }
+}
+
+impl From<OwnableError> for PaymentSplitterError {
+    fn from(ownable: OwnableError) -> Self {
+        match ownable {
+            OwnableError::CallerIsNotOwner => PaymentSplitterError::Custom(String::from("O::CallerIsNotOwner")),
+            OwnableError::NewOwnerIsZero => PaymentSplitterError::Custom(String::from("O::NewOwnerIsZero")),
+        }
+    }
+}
+
+impl From<PausableError> for PaymentSplitterError {
+    fn from(pausable: PausableError) -> Self {
+        match pausable {
+            PausableError::Paused => PaymentSplitterError::Custom(String::from("P::Paused")),
+            PausableError::NoPaused => PaymentSplitterError::Custom(String::from("P::NoPaused")),
+        }
+    }
+}
+
+impl From<ReentrancyGuardError> for PaymentSplitterError {
+    fn from(guard: ReentrancyGuardError) -> Self {
+        match guard {
+            ReentrancyGuardError::ReentrantCall => PaymentSplitterError::Custom(String::from("RG::ReentrantCall")),
```

Comment:
It is optional to use `Ownable` or `AccessControl` in `PaymentSplitter` contract. It is bad to include it by default=) We implements trait `From` for our contract because you can implement trait or, if you defined the trait or if defined the struct. So user outside of open brush will not able to implement `From` for our errors.

If we use `Cutsom` error with `String`, then better to use it everywhere. Because we already included the code fir that error.

Reply:
Ok, I can see that. Still, `RG::ReentrantCall` and those abbreviations look unnecessary to me. But the best I think would be to *not* implement the traits, because if a contract does not use some error - why implement `From`?

Reply:
If developer decides to use some modifiers from our library, then he will need implementation of `From` in this case.